app/controllers/api/v2/hosts_bulk_actions_controller.rb

```diff
@@ -63,6 +63,21 @@ def build
         end
       end
 
+      api :PUT, "/hosts/bulk/reassign_hostgroups", N_("Reassign hostgroups")
+      param_group :bulk_host_ids
+      param :hostgroup_id, :number, :desc => N_("ID of the hostgroup to reassign the hosts to")
+      def reassign_hostgroup
+        hostgroup = params[:hostgroup_id].present? ? Hostgroup.find(params[:hostgroup_id]) : nil
+        BulkHostsManager.new(hosts: @hosts).reassign_hostgroups(hostgroup)
+        if hostgroup
+          process_response(true, { :message => n_("Reassigned #{@hosts.count} host to hostgroup #{hostgroup.name}",
+            "Reassigned #{@hosts.count} hosts to hostgroup #{hostgroup.name}", @hosts.count) })
+        else
+          process_response(true, { :message => n_("Removed assignment of host group from #{@hosts.count} host",
+            "Removed assignment of host group from #{@hosts.count} hosts", @hosts.count) })
```

Comment:
This still looks incorrect to me. I don't think you're supposed to use Ruby interpolation in a translated string.

I think instead of `#{@hosts.count}` it should just be `{count}` without the `#`, as in the [previous comment](https://github.com/theforeman/foreman/pull/10202#discussion_r1643436527).